### PR TITLE
Add examples and add docs

### PR DESCRIPTION
**Pull request by 1-rafael-1**
Hi @barafael , as promised here are a few examples. I got a bit carried away, it was too much fun.
I tested all the examples on my hardware and they seem to do what I expected them to.

I understood way too little of what I was doing so I attempted to read up to understand the driver better. That went over my head so I finally put an agent to work on it an NOW I fell I have a better grasp. 

Downside: Most of what the documentation I added says, I cannot verify. It reads logical but at such a level of skill as I am who knows... You mentioned on matrix to maybe having the same issue. So it boils down to: Do we think the docs improve the situation of code-not-understood opposed to code-that-may-have-wrong-docs-in-some-places. I would tend to think yes... 

But it is Your driver and I will NOT be disappointed if You decide to not want that in. In that case You may want to keep some of the examples, still?

### CODE REVIEW

**Review by barafael — APPROVED** [review]
Really really good stuff!

File: examples/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2024 rafael
```
[the remark this answers is not in the document]
Comment (1-rafael-1) [reply examples/LICENSE:3]: nah, copy&paste error -> will throw it out

File: examples/src/basic_async.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    // This step resets the DMP to its default state and loads the firmware
+    info!("Initializing DMP Firmware");
+    sensor.initialize_dmp(&mut delay).await.unwrap();
+    info!("DMP Firmware Initialized");
+
+    // Read raw accelerometer data (uncalibrated)
+    // The accelerometer measures linear acceleration in three axes (X, Y, Z)
+    // Values will be imprecise until calibration is performed
+    let accel_data = sensor.accel().await.unwrap();
+    info!("Accelerometer Data: {:?}", Debug2Format(&accel_data));
```
[the remark this answers is not in the document]
Comment (1-rafael-1) [reply examples/src/basic_async.rs:66]: #11 already has that, bear with me here :-)
Debug2Format I got taught by James Munns on the Embassy matrix channel, that and 5 other things, great guy.

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# MPU6050-DMP Examples
+
+This directory contains examples demonstrating various features of the MPU6050 sensor with its Digital Motion Processor (DMP).
+
+## Examples
+
+### Basic Async (`src/basic_async.rs`)
```
[the remark this answers is not in the document]
Comment (1-rafael-1) [reply examples/README.md]: agree

**Comment by 1-rafael-1** [comment]
@barafael done

File: examples/LICENSE (hunk 1 above)
[the remark this answers is not in the document]
Comment (1-rafael-1) [reply examples/LICENSE]: deleted

**Pull request merged**